solutions/CSES Planets Cycles.mdx

```diff
@@ -0,0 +1,93 @@
+---
+id: cses-1751
+source: CSES 
+title: Planets Cycles
+author: Maggie Liu
+---
+
+Define the number of teleportations starting from a planet as the $\texttt{cyclelength}$ of that planet. For each planet that hasn't been visited, we want to find its $\texttt{cyclelength}$. Call the planet we are performing $\texttt{dfs}$ from the $\textit{start}$. As we perform $\texttt{dfs}$ from the $\textit{start}$, keep track of the planets seen, in order, in the $\texttt{cycle}$ vector. When we reach a planet that has already been visited (call this planet the $\textit{repeat}$), we add the $\texttt{cyclelength}$ of the $\textit{repeat}$ to the current $\texttt{step}$ count. 
+
+Once we have $\texttt{cycle}$ and $\texttt{steps}$ (the $\texttt{cyclelength}$ of the $\textit{start}$) we can calculate the $\texttt{cyclelength}$ of all the planets in this $\texttt{cycle}$. We know the $\textit{repeat}$ will always be the planet at the end of the $\texttt{cycle}$, but there are two cases for the $\textit{repeat}$. 
+1. The $\textit{repeat}$ was visited twice in the current $\texttt{cycle}$.
+2. The $\textit{repeat}$ was visited previously and only appears at the end of the current $\texttt{cycle}$.
+
+For the first case, let the planets in the $\texttt{cycle}$ between the two occurrences of the $\textit{repeat}$ be the $\textit{loop}$. For the planets inside the $\textit{loop}$, the repeating planet when starting from that planet is itself. Thus, the $\texttt{cyclelength}$ of any planet inside the $\textit{loop}$ will be the same as the $\texttt{cyclelength}$ of the $\textit{repeat}$. For all the planets outside the $\textit{loop}$, the planet that repeats when starting from each planet will still be the $\textit{repeat}$.
+
+For the second case, all the planets are outside the $\textit{loop}$. The repeating planet when starting from any of the planets in the $\texttt{cycle}$ will be the $\textit{repeat}$.
+
+As we iterate through the planets outside the $\textit{loop}$, the $\texttt{cyclelength}$ will decrease by $1$ each time, starting from $\texttt{steps}$. For planets inside the $\textit{loop}$, the $\texttt{cyclelength}$ will be the same as the $\texttt{cyclelength}$ of the $\textit{repeat}$.
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+using namespace std;
+void dfs(int planet);
```

Comment:
Rather than defining the DFS function after the main function, why don't you define it before?

Reply:
I prefer having the main function first but i can move the dfs function.
> Also, could you make your explanation more concise? The explanation you put seems too long.

Do you have any suggestions on what I could remove? It is long but I'm not sure what to cut.

Reply:
> I prefer having the main function first but i can move the dfs function.
> Yeah, this would be better. Most people don't follow the same coding pattern as you.
> > Also, could you make your explanation more concise? The explanation you put seems too long.
> Do you have any suggestions on what I could remove? It is long but I'm not sure what to cut.
> Not specifically, no. Could anyone else help with this? @andrewwangva @frodakcin @HarryWangATX

Reply:
doesn't really matter where it is.